### ISSUE TITLE
git fails on OS X

### ISSUE THREAD

**Issue by certik** [hashdist/hashstack#745]
```
[git] compat/apple-common-crypto.h: In function 'git_CC_EVP_EncodeBlock':
[git] In file included from git-compat-util.h:351:0,
[git]                  from cache.h:4,
[git]                  from ident.c:8:
[git] compat/apple-common-crypto.h: In function 'git_CC_EVP_EncodeBlock':
[git] compat/apple-common-crypto.h:32:2: error: unknown type name 'SecTransformRef'
[git]   SecTransformRef encoder;
[git]   ^
```

**Comment by Mark-E-Hamilton** [comment]
Did you find a resolution for this issue? I just tried building git-2.6.1 from source and got the same error.

**Comment by certik** [comment]
Yes, the solution is to disable compiling the `apple-common-crypto.h` file. I think `export NO_APPLE_COMMON_CRYPTO=1` does the work.

**Comment by Mark-E-Hamilton** [comment]
Thanks, that got me past the compiler errors. Now, however, I'm getting this link error

    LINK git-credential-store
    Undefined symbols for architecture x86_64:
      "_reencode_string_iconv", referenced from:
          _precompose_argv in libgit.a(precompose_utf8.o)
    ld: symbol(s) not found for architecture x86_64
    collect2: error: ld returned 1 exit status
    make: *** [git-credential-store] Error 1

Any thoughts on this?

**Comment by certik** [comment]
Yes, that seems to be a problem with your libiconv, see e.g.: https://trac.macports.org/ticket/38851.

Are you using hashstack to build git? We build our own libiconv, that we bootstrap properly, so it should just work. If you can paste the whole problem, we can have a look.

**Comment by Mark-E-Hamilton** [comment]
Not sure what you mean by the whole problem. We're doing a fairily standard build from source

    tar zxf git-2.6.1.tar.gz
    cd git-2.6.1
    export NO_APPLE_COMMON_CRYPTO=1
    make configure
    make

I did check to see that we had the most recent macports version of libiconv installed (1.14) and we do.

However, I don't know what hashstack is, so it might be that this is the wrong venue for this question. I was really just responding to your original post to see if you had resolved it. I've also posted this question on the git-users google group; perhaps that's a better place for it?

**Comment by certik** [comment]
`hashstack` is this repository where the issue you posted a comment belongs to, it's a source distribution, you can read more about it here:http://hashdist.github.io/. I wasn't sure if you were having problems using it, or if you found the issue by google and are installing things by hand. Thanks for clarifying it.

Yes, asking on the git mailinglist is definitely the way to go.

**Comment by dimpase** [comment]
git-2.4.0 simply does not build on OSX 10.12. High time to upgrade! I checked that git-2.10.2 builds with openssl compiled from source (with --with-openssl= flag set appropriately in ./configure)

see more on this here https://github.com/sagemath/binary-pkg/issues/8
(a @vbraun 's package)...
